### PR TITLE
ARROW-16790: [Go][Parquet] Avoid unnecessary memory allocations for skipping rows

### PR DESCRIPTION
**Pull request by zeroshade**

### CODE REVIEW

**Comment by zeroshade** [comment]
CC @mdepero to take a look as the original reporter

File: go/parquet/file/column_reader.go [hunk 1]
```diff
@@ -136,8 +138,8 @@ type columnChunkReader struct {
 
 // NewColumnReader returns a column reader for the provided column initialized with the given pagereader that will
 // provide the pages of data for this column. The type is determined from the column passed in.
-func NewColumnReader(descr *schema.Column, pageReader PageReader, mem memory.Allocator) ColumnChunkReader {
-	base := columnChunkReader{descr: descr, rdr: pageReader, mem: mem, decoders: make(map[format.Encoding]encoding.TypedDecoder)}
+func NewColumnReader(descr *schema.Column, pageReader PageReader, mem memory.Allocator, bufferPool *sync.Pool) ColumnChunkReader {
```
Comment (pitrou) [inline go/parquet/file/column_reader.go:141]: Shouldn't you add a terse description of the new parameter?
(especially as the type isn't self-descriptive)

Comment (zeroshade) [reply go/parquet/file/column_reader.go:145]: That's a good point, I've added to the comment.

**Review by pitrou — APPROVED** [review]
I can't evaluate the technical details, but the implementation looks sound.

**Comment by mdepero** [comment]
> CC @mdepero to take a look as the original reporter

FWIW all our tests covering previous issues were green after this PR, thanks for the fix! @zeroshade

**Pull request merged**